data/lua/wml/random_placement.lua

```diff
@@ -14,7 +14,7 @@ wesnoth.wml_actions.random_placement = function(cfg)
 	local math_abs = math.abs	
 	local locs = wesnoth.get_locations(filter)
 	if type(num_items) == "string" then		
-		num_items = math.floor(loadstring("local size = " .. #locs .. "; return " .. num_items)())
+		num_items = math.floor(load("local size = " .. #locs .. "; return " .. num_items)())
```

Comment:
I always think about security issues. You get like that if you have to spend a month working with CERT/CC and downstream vendors to publish a security update.

But, you are correct: anyone can do anything they want in a [lua] tag. It's just a bit surprising to find here, and in one AI function, they can do it in a less-than-obvious manner.